api/main.py

```diff
@@ -351,6 +361,9 @@ async def post_node(node: Node, token: str = Depends(get_user)):
                 status_code=status.HTTP_404_NOT_FOUND,
                 detail=f"Parent not found with id: {node.parent}"
             )
+
+    await _verify_user_group_existence(node.user_groups)
```

Comment:
One thought, we could also skip this and instead accept that the owner and group may not always exist in the database.  That way, when renaming or deleting users and groups we don't have to update retroactively all the older nodes.  I think it's easy to handle as the user and group lookups are done directly in the application logic.

Reply:
I don't think inserting invalid data into the database for the sake of simplicity is a good idea.

Reply:
It depends how we define invalid, having an non-existing object ID would be more problematic I think.  But even that we're not checking afaik, say when setting a parent node id.  Ultimately, if the user provides bad data then it's a problem on the user's side.

What's important though is that we define exactly who is in charge of the data consistency.  If we want to check that groups exist and other similar checks around node ownership then we need to do it entirely, and update all nodes when a group or user gets renamed etc.  Otherwise, we can just say it's something the user is in charge of and be fault-tolerant to return HTTP 4xx errors when doing operations on data that doesn't exist.  Having fixed user names and groups is also a way to solve this by not having the use-case where things need to be renamed and updated.